docker/docker_client.go

```diff
@@ -221,10 +221,87 @@ func CheckAuth(ctx context.Context, sCtx *types.SystemContext, username, passwor
 	}
 }
 
+// Data holds the output of both the v2 and v1 endpoints
+type Data struct {
+	// Repositories holds the results returned by the /v2/_catalog endpoint
+	Repositories []string `json:"repositories"`
+	// Results holds the results returned by the /v1/search endpoint
+	Results []Results `json:"results"`
+}
+
+// Results holds the information of each matching image returned by the v1 endpoint
+type Results struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	StarCount   int    `json:"star_count"`
+	IsTrusted   bool   `json:"is_trusted"`
+	IsAutomated bool   `json:"is_automated"`
+	IsOfficial  bool   `json:"is_official"`
+}
+
+// SearchRegistries queries a list of registries for a matching image
+func SearchRegistries(ctx context.Context, sCtx *types.SystemContext, registries []string, image, limit string) (map[string][]Results, error) {
+	searchResults := make(map[string][]Results)
+	for _, reg := range registries {
+		url := reg
+		if strings.Contains(reg, "docker") {
+			url = "index.docker.io"
+		}
+
+		logrus.Debugf("pinging v2 endpoint\n")
+		data, err := getRegistryData(ctx, sCtx, url, image, limit)
+		if err != nil {
+			fmt.Printf("couldn't search registry %q: %v", reg, err)
+			continue
+		}
+
+		if len(data.Results) > 0 {
+			searchResults[reg] = data.Results
+			continue
+		}
+
+		results := []Results{}
+		for _, repo := range data.Repositories {
+			if strings.Contains(repo, image) {
```

Comment:
Should this respect `limit`? (I don’t have a strong opinion.)

Reply:
The limit is only for the v1 endpoint. I am changing the output in podman based on the number of results I get. Could change the variable to `v1Limit`, however the limit query doesn't always work. For example it doesn't work for the redhat registry.

Reply:
Fair enough, we can let the caller deal with excess results. (Or we could truncate them in this function, for both v1 and v2; both would be consistent. But throwing away received data seems wasteful.)